leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -205,4 +215,81 @@ private void markAsNotLeading(final LeadershipToken leadershipToken, @Nullable T
         throw notCurrentLeaderException("method invoked on a non-leader (leadership lost)", cause);
     }
 
+    /**
+     * Wraps a delegate Logger, prepending the specified string to the start of each log message.
+     */
+    private static class TaggedLogger {
+        private final Logger delegate;
+        private final String tag;
+
+        public TaggedLogger(Class clazz, String tag) {
+            this.delegate = LoggerFactory.getLogger(clazz);
+            this.tag = "[" + tag + "] ";
+        }
+
+        public TaggedLogger(String name, String tag) {
+            this.delegate = LoggerFactory.getLogger(name);
+            this.tag = "[" + tag + "] ";
```

Comment:
might suggest making simple constructor `TaggedLogger(Logger, String)` and make these static factories or delegate to that.